common/persistence/sql/sqlplugin/postgres/plugin.go

```diff
@@ -73,6 +73,23 @@ func (d *plugin) CreateAdminDB(cfg *config.SQL) (sqlplugin.AdminDB, error) {
 	return db, nil
 }
 
+func composeConnectionString(user, password, host, port, dbName string) string {
+	composeSegment := func(paramName string, paramValue string) string {
+		paramValue = strings.TrimSpace(paramValue)
+		if paramValue != "" {
+			return paramName + "=" + paramValue + " "
+		}
+		return ""
+	}
+
+	return composeSegment("user", user) +
```

Comment:
I would suggest to use something like string builder here, but we call this once, right? So it doesn't really matter.

Reply:
Ah, yes, good point, my assumption is that the cost of this is negligible compared to the cost of actual db interactions / network (and i *think* this gets executed once / or "rarely", when a db connection needs to be reestablished -- although I didn't profile verify this assumption), so I optimized this for dev experience (simplicity / readability).